Formula/php.rb

```diff
@@ -0,0 +1,362 @@
+      %w[EXTRA_LDFLAGS EXTRA_LDFLAGS_PROGRAM].each do |mk_var|
+        system_libs = []
+        other_flags = []
+        s.get_make_var(mk_var).split.each do |f|
+          if f[%r{^-L\/(?:Applications|usr\/lib)\/}]
+            system_libs << f
+            next
+          end
+          other_flags << f
+        end
+        s.change_make_var! mk_var, other_flags.concat(system_libs).join(" ")
+      end
+    end
+
+    system "make"
+    ENV.deparallelize
+    system "make", "install"
+
+    config_path.install "php.ini-development" => "php.ini"
+    (config_path/"php-fpm.d").install "sapi/fpm/www.conf"
+    config_path.install "sapi/fpm/php-fpm.conf"
+    inreplace config_path/"php-fpm.conf", /^;?daemonize\s*=.+$/, "daemonize = no"
+
+    unless File.exist? "#{var}/log/php-fpm.log"
+      (var/"log").mkpath
+      touch var/"log/php-fpm.log"
+    end
+  end
+
+  def caveats
+    s = []
+
+    s << <<-EOS.undent
+      To enable PHP in Apache add the following to httpd.conf and restart Apache:
+          LoadModule php7_module #{HOMEBREW_PREFIX}/lib/httpd/modules/libphp7.so
+
+          <FilesMatch \.php$>
+              SetHandler application/x-httpd-php
+          </FilesMatch>
+
+      Finally, check DirectoryIndex includes index.php
+          DirectoryIndex index.php index.html
+    EOS
+
+    s << <<-EOS.undent
+      The php.ini and php-fpm.ini file can be found in:
+          #{etc}/php/#{php_version}/
+
+      The snmp extension is disabled by default since it crashes when used with the Apache SAPI.
+    EOS
+
+    s.join "\n"
+  end
+
+  def post_install
+    chmod 0755, lib/"php/.channels"
+    chmod 0755, lib/"php/.channels/.alias"
+    chmod 0644, (Dir.glob(lib/"php/.channels/**/*", File::FNM_DOTMATCH).reject { |a| a =~ %r{\/\.{1,2}$} || File.directory?(a) })
+    chmod 0644, %W[
+      #{lib}/php/.depdblock
+      #{lib}/php/.filemap
+      #{lib}/php/.depdb
+      #{lib}/php/.lock
+    ]
+
+    php_basename = File.basename `#{bin}/php-config --extension-dir`.chomp
+    php_ext_dir = opt_prefix/"lib/php/extensions/" + php_basename
+
+    # fix pear config to use opt paths
+    php_lib_path = opt_lib/"php"
+    {
+      "php_ini" => etc/"php/#{php_version}/php.ini",
+      "php_dir" => php_lib_path,
+      "ext_dir" => php_ext_dir,
+      "doc_dir" => php_lib_path/"doc",
+      "bin_dir" => opt_bin,
+      "data_dir" => php_lib_path/"data",
+      "cfg_dir" => php_lib_path/"cfg",
+      "www_dir" => php_lib_path/"htdocs",
+      "man_dir" => php_lib_path/"local/man",
+      "test_dir" => php_lib_path/"test",
+      "php_bin" => opt_bin/"php",
+    }.each do |key, value|
+      system bin/"pear", "config-set", key, value, "system"
+    end
+
+    %w[
+      ldap
+      imap
+      opcache
+      snmp
+    ].each do |e|
+      ext_config_path = (etc/"php/#{php_version}/conf.d/ext-#{e}.ini")
+      extension_type = (e == "opcache") ? "zend_extension" : "extension"
+      # Do not load snmp extension by default
```

Comment:
This should be moved to the else of the `exist?` check, otherwise it will cause an error when users upgrade and have manually enabled the module.